lib/set.js

```diff
@@ -59,13 +67,12 @@ Set.prototype.addMigration = function(title, up, down){
  */
 
 Set.prototype.save = function(fn){
-  var self = this
-    , json = JSON.stringify(this);
-  fs.writeFile(this.path, json, function(err){
-    if (err) return fn(err);
-
+  var self = this;
+  this.state.save({
```

Comment:
One possible resolution to the migration conflict problem (i.e. two developers both adding a migration at the same time) is to maintain an index file containing an ordered list of all migrations.

If added to the repo, version control will catch the conflict before any merging happens. It would be simple to extend the create action to maintain this index file.